engine/render/src/render/font_renderer.cpp

```diff
@@ -700,6 +701,31 @@ namespace dmRender
         int line_count = Layout(text, width, lines, max_lines, &layout_width, lm, measure_trailing_space);
         float x_offset = OffsetX(te.m_Align, te.m_Width);
         float y_offset = OffsetY(te.m_VAlign, te.m_Height, font_map->m_MaxAscent, font_map->m_MaxDescent, te.m_Leading, line_count);
+        float layout_height = line_count * (line_height * te.m_Leading) - line_height * (te.m_Leading - 1.0f);
+
+        // find X,Y local coordinate of text center
+        float center_x = x_offset; // start from the X position of the pivot point
+        switch (te.m_Align) {
+            case TEXT_ALIGN_LEFT:
+                center_x += layout_width/2; // move halfway to the right since we're aligning left
+            break;
+            case TEXT_ALIGN_RIGHT:
+                center_x -= layout_width/2; // move halfway to the left from pivot since we're aligning right
+            break;
+            // nothing to do for TEXT_ALIGN_CENTER. Pivot is already at the center of the text X-wise
+        }
+        float center_y = y_offset + font_map->m_MaxAscent - layout_height/2; // 'y_offset' to move to the baseline of first letter in text. +'MaAscent' to get to the top of the text. -layout_height to get to the center.
+
+        // find center and radius for frustum culling
+        dmVMath::Vector4 centerpoint_local(center_x, center_y, 0, 1);
+        dmVMath::Vector4 cornerpoint_local(centerpoint_local.getX() + layout_width/2, centerpoint_local.getY() + layout_height/2, centerpoint_local.getZ(), centerpoint_local.getW());
+        dmVMath::Vector4 centerpoint_world = te.m_Transform * centerpoint_local; // transform to world coordinates
+        dmVMath::Vector4 cornerpoint_world = te.m_Transform * cornerpoint_local;
+
+        te.m_FrustumCullingRadius = Vectormath::Aos::length(cornerpoint_world - centerpoint_world);
+        te.m_FrustumCullingCenter.setX(centerpoint_world.getX());
+        te.m_FrustumCullingCenter.setY(centerpoint_world.getY());
+        te.m_FrustumCullingCenter.setZ(centerpoint_world.getZ());
```

Comment:
This is the wrong place to do it.
At this point in time, we are past the frustum culling, and we are generating the vertex data for rendering.

You need to figure out the info _before_ the frustum culling function is called.
You should do it in `DrawText`, when the text is registered (just as you had it before).